File: erlang/ch1/ex112.erl
%%%-------------------------------------------------------------------
%%% @author Yohan
%%% @copyright (C) 2018, <COMPANY>
%%% @doc
%%% compile with > c("exXXX.erl").
%%% load with
%%% > cd("c").
%%% > l(exXXX).
%%% run with > exXXX:main().
%%% %%% @end
%%% Created : 19. Aug 2018 12:30 PM
%%%-------------------------------------------------------------------
-module(ex112).
-author("Yohan").

%% API
-export([main/0, pascal/2]).

main() ->
  io:fwrite("~w\n", [pascal(6, 1)]),
  io:fwrite("~w\n", [pascal(6, 2)]),
  io:fwrite("~w\n", [pascal(6, 3)]),
  io:fwrite("~w\n", [pascal(6, 4)]),
  io:fwrite("~w\n", [pascal(6, 5)]),
  io:fwrite("~w\n", [pascal(6, 6)]).

pascal(_, 1) -> 1;
pascal(R, N) when N == R -> 1;
pascal(R, N) -> pascal(R - 1, N - 1) + pascal(R - 1, N).

File: julia/ch1/Ex112.jl
"""
# module Ex112

- Julia version: 0.6.4
- Author: Yohan
- Date: 2018-08-19

# Examples

```jldoctest
julia> include("c:/Users/Yohan/IdeaProjects/sicp/julia/ch1/ex112.jl")

julia> Ex112.pascal(3, 2)
```
"""
module Ex112

pascal(r, n) = (n == 1 || n == r) ? 1 : pascal(r - 1, n - 1) + pascal(r - 1, n)

end